components/core/src/package/ident.rs

```diff
@@ -33,7 +33,8 @@ pub trait Identifiable: fmt::Display + Into<PackageIdent> {
     fn fully_qualified(&self) -> bool { self.version().is_some() && self.release().is_some() }
 
     fn valid(&self) -> bool {
-        let re = Regex::new(r"^[A-Za-z0-9_.-]+$").unwrap();
+        // allow periods in ident name but not at start or end
+        let re = Regex::new(r"^[A-Za-z0-9_-]+([A-Za-z0-9_.-]+[A-Za-z0-9_-]+)?$").unwrap();
```

Comment:
same non-capturing logic applies here, too.

If there were a way to factor this so we could incorporate this regex into the bigger one, it would be great to avoid the duplication.